### ISSUE TITLE
Remove event listeners in `{{x-playback}}` component

### ISSUE THREAD

**Issue by hugurp** [internet4000/radio4000#32]
Not sure how to remove the event listeners in `services/player.js/@didDestroyElement`.

When they are registered (`@didInsertElement`) , their function call are anonymous functions to preserve the `this` context.

To remove them events, as far as I understand, we have to pass their handler function name to the `removeEventListener` function — and it can't be done in our current setup since they were registered anonymously.

I could not find another way to register them, failed using `.bind(this)` on a named function; as well as for every solutions described here:
[https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener)

#needhelp

**Comment by oskarrough** [comment]
Good question. I believe `Ember.service` is a singleton and not destroyed until you actually close the browser tab.

But it does have https://www.emberjs.com/api/ember/2.14/classes/Ember.Service/methods/willDestroy?anchor=willDestroy

Perhaps we should move it to the `playback` component? Or did you find away the last 19 hours? 🥇

**Comment by hugurp** [comment]
I actually tried to remove the `player.service`.
What happened:
- the `track.route` is where order is first given to play a track
- in our actual implementation that triggers the `@playTrack` in `player.service`
- so i wanted to replace that by bubbling up a `playTrack` action up to the `application.route`, and there set the `track` in its `application.controller`, which then would be passed down to `{{app-root track}}` and to `{{x-playback track}}`

I think that solution would be better (one service less), buuuuuuut: an action bubbled up from track route is never passed to the `application.controller` (so never to the template, and therefore not to the components we need the data in).

I did not wanted to do some magic tricks reloading the route, or setting the track data from the `application.route` to the `application.controller`.

Conclusion:
I dicided it was easier to keep the `player.service`, and now I think it is actually the right decision because it also makes it a nice abstract place to take care of `active.track/channel` manipulation, when the `{{x-playback}}` takes more care about the actual implementation and direct DOM-related functions of the `radio4000-player-vue`.

Correction:
My issue was a mistake, it is not in the `player.service` that the event listeners are registered but in `{{x-playback}}` componenet. My mistake, I edit the issue RN.
